src/components/Asset/AssetActions/Download.tsx

```diff
@@ -53,16 +63,42 @@ export default function Download({
     setValidOrderTx(asset?.accessDetails?.validOrderTx)
     // get full price and fees
     async function init() {
+      // we only show the gas estimation if the user is connected
```

Comment:
this sounds like it should rather be handled in its own effect after this one, which also would make this init method a bit less complex and we nicely separate concerns